### PR TITLE
Schema revision

### PR DESCRIPTION
**Pull request by danixeee**
**Description of the changes being introduced by the pull request**:

This PR modifies _formats_ in order to support _GPG_ keys.
Signature verification function is also modified to allow passing _GPG_ public key format.

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by danixeee** [comment]
@lukpueh I saw you merged _gpg_ work to master branch, so I have rebased this branch.

**Comment by lukpueh** [comment]
@danixeee, I fear your rebase resurrected some TUF-specific schemas that we removed purposefully with #165.

**Comment by danixeee** [comment]
@lukpueh, sorry for that, I think it should be fine now.

File: securesystemslib/formats.py [hunk 1]
```diff
@@ -408,6 +374,71 @@ def _create_gpg_pubkey_with_subkey_schema(pubkey_schema):
...
+    VERIFICATION_KEY_DICT_SCHEMA])
+
+ANY_PUBKEY_SCHEMA = SCHEMA.OneOf([PUBLIC_KEY_SCHEMA, GPG_PUBKEY_SCHEMA])
+
+ANY_PUBKEY_DICT_SCHEMA = SCHEMA.DictOf(
+  key_schema = KEYID_SCHEMA,
+  value_schema = ANY_PUBKEY_SCHEMA)
+
+ANY_STRING_SCHEMA = SCHEMA.AnyString()
+LIST_OF_ANY_STRING_SCHEMA = SCHEMA.ListOf(ANY_STRING_SCHEMA)
```
Comment (lukpueh) [inline securesystemslib/formats.py]: Any reason why `ANY_STRING_SCHEMA` and `LIST_OF_ANY_STRING_SCHEMA` were moved down here?

File: securesystemslib/keys.py [hunk 2]
```diff
@@ -836,10 +836,10 @@ def verify_signature(key_dict, signature, data):
   # This check will ensure 'key_dict' has the appropriate number
   # of objects and object types, and that all dict keys are properly named.
   # Raise 'securesystemslib.exceptions.FormatError' if the check fails.
-  securesystemslib.formats.ANYKEY_SCHEMA.check_match(key_dict)
+  securesystemslib.formats.ANY_VERIFICATION_KEY_SCHEMA.check_match(key_dict)
 
   # Does 'signature' have the correct format?
-  securesystemslib.formats.SIGNATURE_SCHEMA.check_match(signature)
+  securesystemslib.formats.ANY_SIGNATURE_SCHEMA.check_match(signature)
```
[the remark this answers is not in the document]
Comment (danixeee) [reply securesystemslib/keys.py]: @lukpueh I thought it's fine to get `TypeError` until it is implemented. I have reverted those changes.

File: securesystemslib/formats.py (hunk 1 above)
Comment (danixeee) [reply securesystemslib/formats.py]: It was the latest defined schema in the file, I moved it to the top of the file now.

**Pull request merged**